build/Makefile.openssl

```diff
@@ -7,6 +7,6 @@ $(INSTALL_DIR)/bin/openssl: Makefile
 
 Makefile: config Configure Makefile.org
 	# Force a 64 bit build on MacOS
-	KERNEL_BITS=64 ./config --openssldir=$(INSTALL_DIR) no-shared enable-static-engine -fPIC
+	KERNEL_BITS=64 ./config --install_prefix=$(INSTALL_DIR) --prefix=/ --libdir=lib --openssldir=/etc/ssl no-shared enable-static-engine -fPIC
```

Comment:
Turns out the OSX system version of OpenSSL (0.9.8zg apparently) has some magic Apple pixie dust which allows it to reach into the keychain to retry failed verification attempts with whatever roots it has in there.

Our build of vanilla OpenSSL 1.0.2 can't do that so people are going to have to do it the 'normal' OpenSSL way on OS X too.
Where 'normal' is one of:
   * use the flag, Luke
   * set SSL_CERT_FILE
   * dump some certs in /System/Library/OpenSSL/certs

I added some stuff in the `README.MacOS` doc about it.